src/network/encodable.rs

```diff
@@ -435,6 +457,28 @@ mod tests {
         assert_eq!(serialize(&VarInt(0xF0F0F0F0F0E0)).ok(), Some(vec![0xFFu8, 0xE0, 0xF0, 0xF0, 0xF0, 0xF0, 0xF0, 0, 0]));
     }
 
+    #[test]
+    fn deserialize_nonminimal_vec() {
+        assert!(deserialize::<Vec<u8>>(&[0xfd, 0x00, 0x00]).is_err());
```

Comment:
You need to check the Err type, as these will fail whether its a non-minimal encoding or because of a short-read.